packages/prover/src/interfaces.ts

```diff
@@ -0,0 +1,51 @@
+import {ChainForkConfig} from "@lodestar/config";
+import {NetworkName} from "@lodestar/config/networks";
+import {ProofProvider} from "./proof_provider/proof_provider.js";
+import {ELRequestPayload, ELResponse} from "./types.js";
+
+export enum LCTransport {
+  Rest = "Rest",
+  P2P = "P2P",
+}
+
+export type RootProviderInitOptions = {
+  network: NetworkName;
+  signal: AbortSignal;
+  config?: ChainForkConfig;
+  wsCheckpoint?: string;
+} & ({transport: LCTransport.Rest; urls: string[]} | {transport: LCTransport.P2P; bootnodes: string[]});
```

Comment:
lets keep the same name of `urls` or `bootnodes` in cli/options and here as well so as to just have a clear correspondence that they are all the same